buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -0,0 +1,1297 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.gradle
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.quality.Checkstyle
+import org.gradle.api.plugins.quality.FindBugs
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.tasks.Exec
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.api.tasks.testing.Test
+import org.gradle.testing.jacoco.tasks.JacocoReport
+
+/** This plugin adds methods to configure a module with Beam's defaults, called "natures".
+ *
+ * The natures available:
+ *
+ * <ul>
+ *   <li>Java   - Configures plugins commonly found in Java projects
+ *   <li>Go     - Configures plugins commonly found in Go projects
+ *   <li>Docker - Configures plugins commonly used to build Docker containers
+ *   <li>Grpc   - Configures plugins commonly used to generate source from protos
+ *   <li>Avro   - Configures plugins commonly used to generate source from Avro specifications
+ * </ul>
+ *
+ * <p>For example, see applyJavaNature.
+ */
+class BeamModulePlugin implements Plugin<Project> {
+
+	// A class defining the set of configurable properties accepted by applyJavaNature
+	class JavaNatureConfiguration {
+		double javaVersion = 1.8
+		// Controls the JDK source language and target compatibility
```

Comment:
move comments to the line before instead of after

Reply:
Done